llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8407,10 +8407,13 @@ VPRecipeBuilder::tryToWidenMemory(Instruction *I, ArrayRef<VPValue *> Operands,
     if (Reverse)
       VectorPtr = new VPReverseVectorPointerRecipe(
           Ptr, &Plan.getVF(), getLoadStoreType(I),
-          GEP ? GEP->isInBounds() : false, I->getDebugLoc());
+          GEP && GEP->isInBounds() ? GEPNoWrapFlags::inBounds()
+                                   : GEPNoWrapFlags::none(),
```

Comment:
I kept the previous inbounds-only logic here because propagating nuw for reverse iteration (using negative offsets) seems problematic.